pints/toy/_toy_classes.py

```diff
@@ -61,3 +79,73 @@ def suggested_times(self):
         """
         raise NotImplementedError
 
+    def _rhs(self, y, t, p):
+        """
+        Returns RHS of ODE for numerical integration to obtain outputs and
+        sensitivities. This should be a vector of length ``n_outputs``.
+        """
+        raise NotImplementedError
+
+    def _rhs_S1(self, y_and_dydp, t, p):
+        """
+        Forms the RHS of ODE for numerical integration to obtain both
+        outputs and sensitivities. This should be a vector of length
+        ``n_outputs`` + ``n_parameters``.
+        """
+        y = y_and_dydp[0:self.n_outputs()]
+        dydp = y_and_dydp[self.n_outputs():].reshape((self.n_outputs(),
+                                                      self.n_parameters()))
+        dydt = self._rhs(y, t, p)
+        d_dydp_dt = (
+            np.matmul(self.jacobian(y, t, p), dydp) +
+            self._dfdp(y, t, p))
+        return np.concatenate((dydt, d_dydp_dt.reshape(-1)))
+
+    def simulate(self, parameters, times):
+        """ See :meth:`pints.ForwardModel.simulate()`. """
+        return self._simulate(parameters, times, False)
+
+    def _simulate(self, parameters, times, sensitivities):
```

Comment:
This is all very useful and a Good Idea, but specific to ODE models and could be useful to users implementing new toy ODE models.

Should we have a subclass called e.g. `ODEToyModel` that has all these methods, possibly public, and has more documentation on how to use this functionality?

Reply:
True, yes, you may be right that it's better to create a new class (I was mistakenly thinking that we just had ODE toy models, whereas we have others too).

Reply:
What do you think @martinjrobins ?

Reply:
I'm resolving this as think we decided to have a ToyODEModel class